src/main/java/ru/yandex/practicum/filmorate/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,30 @@
+package ru.yandex.practicum.filmorate.exception;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@Slf4j
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+    @ExceptionHandler(NotFoundException.class) // К сожалению, решение через @ResponseStatus приводит к ошибке 500 вместо 404
```

Comment:
Ты точно уверен, что добавляешь нужный статус ответа в `@ResponseStatus`?
Данный метод должен быть таким:

```suggestion
    @ExceptionHandler(NotFoundException.class)
    @ResponseStatus(HttpStatus.NOT_FOUND)
    public ErrorResponse handleNotFoundException(final NotFoundException e) {
        return new ErrorResponse(e.getMessage());
    }
```